sdk/keyvault/Azure.Security.KeyVault.Certificates/src/CertificateClient.cs

```diff
@@ -191,17 +193,23 @@ public virtual Response<X509Certificate2> DownloadCertificate(string certificate
 
                 if (string.IsNullOrEmpty(value))
                 {
-                    throw new InvalidOperationException($"Secret {certificate.SecretId} contains no value");
+                    throw new InvalidDataException($"Secret {certificate.SecretId} contains no value");
                 }
 
-                if (secret.ContentType == CertificateContentType.Pem)
+                if (secret.ContentType is null || secret.ContentType == CertificateContentType.Pkcs12)
                 {
-                    throw new NotSupportedException($"PEM-formatted certificates are not supported");
-                }
+                    byte[] rawData = Convert.FromBase64String(value);
 
-                byte[] rawData = Convert.FromBase64String(value);
+                    X509Certificate2 x509 = new X509Certificate2(rawData);
+                    return Response.FromValue(x509, secretResponse.GetRawResponse());
+                }
+                else if (secret.ContentType == CertificateContentType.Pem)
+                {
+                    X509Certificate2 x509 = PemReader.LoadCertificate(value.AsSpan(), certificate.Cer, allowCertificateOnly: true);
```

Comment:
It seems PemReader only supports RSA certificates at the moment. This wasn't a concern with ClientCertificateCredentials because ATM AAD only supports RSA certificates for authentication. But the certificate here could also be an EC certificate no? Perhaps we should guard against this and file an issue to support EC certificates as well?

Reply:
This - and the old implementation - only support labels like "BEGIN PRIVATE KEY", not "BEGIN RSA PRIVATE KEY" or "BEGIN EC PRIVATE KEY", so I don't think there's anything to guard as it just won't parse into a valid private key, which will throw an `InvalidDataException` as the old impl did before.

Still, I can open a bug to add support for "RSA PRIVATE KEY" since it could work alternatively.

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/19046

Reply:
I also opened https://github.com/Azure/azure-sdk-for-net/issues/19047 to track support EC private keys for Key Vault.

Reply:
I was thinking that since you're calling get certificate you might have information on what the key type of the current certificate is (although it might just be on the policy which could be different from the actual cert). If this is a case you could get use it to give a better error message.

Reply:
Yeah, it's only in the `key_props` of the policy, so it wouldn't help.

That said, investigations show that Key Vault and `openssl pkcs12` both produce PEM files with only `CERTIFICATE` and `PRIVATE KEY` (see https://github.com/Azure/azure-sdk-for-net/issues/19047#issuecomment-786338654). We should probably add some tests, but I'm not sure there's really any support to add right now - not for Key Vault, at least. Might if you want to support loading full chains (up to and/or including the root) for `ClientCertificateCredential`.